subt/octomap.py

```diff
@@ -154,10 +154,27 @@ def frontiers(img, start, draw=False):
         plt.axes().set_aspect('equal', 'datalim')
         plt.show()
 
-    driveable = white[:1023, :1023] | white[1:, :1023] | white[1:, 1:] | white[:1023, 1:]
+    driveable = white[:, :]
+
+    # use 1 pixel surounding
+    d = driveable[:, :]
+    d2 = d[2:, :] & d[1:-1, :] & d[:-2, :]
+    d3 = d2[:, 2:] & d2[:, 1:-1] & d2[:, :-2]
+    z = np.zeros((1022, 1), dtype=np.bool)
+    d = np.hstack([z, d3, z])
+    z = np.zeros((1, 1024), dtype=np.bool)
+    driveable = np.vstack([z, d, z])
+
+    img[driveable, 0] = 128  # gray
+    img[driveable, 1] = 128
+    img[driveable, 2] = 128
+
     i = np.argmax(score)
     limit_score = 3*max(score)/4
-    goals = [(xy[1][i], xy[0][i]) for i in range(len(xy[0])) if score[i] > limit_score]
+    goals = []
+    for dx in range(-1, 2):
+        for dy in range(-1, 2):
+            goals.extend([(xy[1][i] + dx, xy[0][i] + dy) for i in range(len(xy[0])) if score[i] > limit_score])
```

Comment:
Why do you add nine destinations instead of one?

Reply:
If you plan "safe" route it would never go to the "border"/frontier, which is not already "safe". This was dirty workaround which I did not like but it restored the graph connectivity.

Reply:
Uh. OK :-)
Can you add a comment to the code that explains this?